nixos/modules/virtualisation/containerd.nix

```diff
@@ -27,7 +40,15 @@ in
   config = lib.mkIf cfg.enable {
     virtualisation.containerd.args.config = lib.mkIf (cfg.configFile != null) (toString containerdConfigChecked);
 
-    environment.systemPackages = [pkgs.containerd];
+    virtualisation.containerd.configFile = lib.mkDefault (settingsFormat.generate "containerd.toml"
+      config.virtualisation.containerd.settings);
+
+    virtualisation.containerd.settings = {
+      plugins.cri.containerd.snapshotter = lib.mkIf config.boot.zfs.enabled "zfs";
+      plugins.cri.cni.bin_dir = lib.mkDefault "${pkgs.cni-plugins}/bin";
```

Comment:
We might need to spent a few seconds on how potentially resolve for alternative snapshotters. NOt that I mean we should address that here, fully. But at least form an idea hat to do if both, `zfs` and e.g. `stargz` are enabled.

Reply:
Do we have `stargz` packaged at all? Otherwise I would stick to zfs as a reasonable default.

Reply:
> Do we have stargz packaged at all?

Not yet. :wink: 

> Users should be able to override this if say use zfs and want to use a different snapshot.

Perfect for now!